pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1237,6 +1249,30 @@ cdef _broadcast_floordiv_td64(int64_t value, object other,
         return res
 
 
+cdef _require_td64_has_unit(value):
+    """
+    Require that a timedelta64 scalar have a unit specified,
+    i.e. 'timedelta64[ns]' is OK but 'timedelta64' is not.
+
+    Parameters
+    ----------
+    value : np.timedelta64 or ndarray[timedelta64]
+
+    Raises
+    ------
+    ValueError
+    """
+    # GH#19388
+    if value.dtype == 'm8':
+        # i.e. has no unit
+        if util.is_array(value) or (value.view('i8') != NPY_NAT):
```

Comment:
I think you can just do a direct check here

```
In [1]: np.timedelta64('NaT')
Out[1]: numpy.timedelta64('NaT')

In [2]: np.timedelta64('NaT') is np.timedelta64(1, 'ns')
Out[2]: False

In [3]: np.timedelta64('NaT') is np.array([1], dtype='m8[ns]')
Out[3]: False
```

IOW *only*  if it IS ``np.timedelta64('NaT')`` will it be  ok

Reply:
Doesn't work like `pd.NaT`:

```
>>> np.timedelta64('NaT') is np.timedelta64('NaT')
False
```